framework/include/utils/RankFourTensor.h

```diff
@@ -442,10 +439,6 @@ struct RawType<RankFourTensorTempl<T>>
 };
 }
 
-typedef RankFourTensorTempl<Real> RankFourTensor;
-typedef RankFourTensorTempl<DualReal> DualRankFourTensor;
```

Comment:
Go ahead and put this into the forward declaration header as well. We may still need this typedef available for backwards compatibility with applications. Equivalent comment for the other `Ranks`

Reply:
Done. The only occurrence of DualRankTwoTensor is in unit test, and I have changed that to ADRankTwoTensor. Not sure if someone should put a deprecation warning in DualRankTwoTensor.